include/fiction/algorithms/simulation/sidb/calculate_energy_and_state_type.hpp

```diff
@@ -0,0 +1,89 @@
+//
+// Created by Jan Drewniok on 20.04.23.
+//
+
+#ifndef FICTION_CALCULATE_ENERGY_AND_STATE_TYPE_HPP
+#define FICTION_CALCULATE_ENERGY_AND_STATE_TYPE_HPP
+
+#include "fiction/algorithms/simulation/sidb/energy_distribution.hpp"
+#include "fiction/technology/charge_distribution_surface.hpp"
+#include "fiction/utils/math_utils.hpp"
+
+#include <cstdint>
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace fiction
+{
+/**
+ *  Types of excited charge distributions.
+ */
+enum class state_type
```

Comment:
Sorry to bother you with the same comment so often: `state_type` seems not descriptive enough for this type name. Can we find something more fitting?

Also, it should be part of the RST documentation.